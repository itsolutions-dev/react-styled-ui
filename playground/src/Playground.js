// @flow

import React from 'react';
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live';
import ThemeProvider from '../../src/ThemeProvider';
import theme from '../../src/theme';

const getCode = code => `
  <ThemeProvider theme={theme}>
    ${code}
  </ThemeProvider>
`.trim().replace(/^/, '\t');

type PlaygroundProps = {
  code: string,
  scope: Object
};

export default ({ code, scope, ...others }: PlaygroundProps) =>
  (<LiveProvider
    code={getCode(code)}
    scope={{
      ThemeProvider,
      theme,
      ...scope,
    }}
    {...others}
  >
    <LiveEditor />
    <LiveError />
    <LivePreview />
  </LiveProvider>);
