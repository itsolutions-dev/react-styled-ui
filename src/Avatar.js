// @flow

import React from 'react';
import styled, { css } from './styledComponent';
import {
  getSizeOrDefault,
} from './utils/';

const defaultSize = '40px';
const defaultRadius = '99999px';

type AvatarProps = {
  src: string,
  alt: string,
  children: any,
};

const Avatar = styled({
  component: 'div',
  props: {
    children: ({ src, alt, children }: AvatarProps) => (
      src == null ? children : <img src={src} alt={alt || src} />
    ),
  },
  style: css`    
    ${props => (props.src == null ? `
      width: ${getSizeOrDefault(props.size, defaultSize)};
      height: ${getSizeOrDefault(props.size, defaultSize)};
      border-radius: ${getSizeOrDefault(props.radius, defaultRadius)}; 
      color: ${(props.color == null ? 'rgb(255, 255, 255)' : props.color)};
      background-color: ${(props.backgroundColor == null ? 'rgb(188, 188, 188)' : props.backgroundColor)} !important;
      font-size: ${getSizeOrDefault(props.fontSize, '20px')};
      user-select: none;
      align-items: center;
      justify-content: center;
      display: inline-flex;
    ` : `
      > img {
        width: ${getSizeOrDefault(props.size, defaultSize)};
        height: ${getSizeOrDefault(props.size, defaultSize)};
        border-radius: ${getSizeOrDefault(props.radius, defaultRadius)};
        display: inline-flex;
      } 
    `)} 
  `,
});

export default Avatar;
