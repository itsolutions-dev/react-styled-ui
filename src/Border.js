import styled, { css } from './styledComponent';
import {
  getBorder,
  getSizeOrDefault,
} from './utils/';

const defaultWidth = '1px';

const Border = styled({
  component: 'div',
  style: css`
    border-style: ${props => props.borderStyle};
    border-top-width: ${props => props.top !== undefined ? getSizeOrDefault(props.top, defaultWidth) : 0};
    border-bottom-width: ${props => props.bottom !== undefined ? getSizeOrDefault(props.bottom, defaultWidth) : 0};
    border-left-width: ${props => props.left !== undefined ? getSizeOrDefault(props.left, defaultWidth) : 0};
    border-right-width: ${props => props.right !== undefined ? getSizeOrDefault(props.right, defaultWidth) : 0};
    border-color: ${props => props.color !== undefined ? props.color : getBorder(props)};
    color: inherit !important;
  `,
});

Border.defaultProps = {
  borderStyle: 'solid',
};

export default Border;
