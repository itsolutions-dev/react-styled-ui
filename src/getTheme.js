import defaultTheme from './theme';
import deepMerge from './utils/deepMerge';

const getTheme = deepMerge.bind(null, defaultTheme);

export default getTheme;
