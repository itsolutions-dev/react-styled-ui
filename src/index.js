export Absolute from './Absolute';
export Avatar from './Avatar';
export BackgroundImage from './BackgroundImage';
export Badge from './Badge';
export Blockquote from './Blockquote';
export Border from './Border';
export Button from './Button';
export ButtonCircle from './ButtonCircle';
export ButtonGroup from './ButtonGroup';
export Card from './Card';
export Checkbox from './Checkbox';
export Code from './Code';
export Container from './Container';
export Divider from './Divider';
export Fixed from './Fixed';
export Footer from './Footer';
export getTheme from './getTheme';
export Heading from './Heading';
export Heading1 from './Heading1';
export Heading2 from './Heading2';
export Heading3 from './Heading3';
export Heading4 from './Heading4';
export Heading5 from './Heading5';
export Heading6 from './Heading6';
export Image from './Image';
export InlineText from './InlineText';
export Input from './Input';
export Label from './Label';
export Link from './Link';
export List from './List';
export ListItem from './ListItem';
export Loader from './Loader';
export Message from './Message';
export Overlay from './Overlay';
export Progress from './Progress';
export Radio from './Radio';
export Relative from './Relative';
export Select from './Select';
export Slider from './Slider';
export Text from './Text';
export Textarea from './Textarea';
export Toggle from './Toggle';
export Toolbar from './Toolbar';
export Tooltip from './Tooltip';
export Truncate from './Truncate';
export ThemeProvider from './ThemeProvider';
export theme from './theme';
