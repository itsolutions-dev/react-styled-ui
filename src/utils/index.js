export deepMerge from './deepMerge';
export getOpacity from './getOpacity';
export getProperty from './getProperty';
export getHover from './getHover';
export getBackgroundColor from './getBackgroundColor';
export getTextColor from './getTextColor';
export getBorder from './getBorder';
export getHeight from './getHeight';
export hexToRgb from './hexToRgb';
export getColorWithAlpha from './getColorWithAlpha';
export getSizeOrDefault from './getSizeOrDefault';
export guid from './guid';
