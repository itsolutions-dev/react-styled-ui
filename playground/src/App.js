import React from 'react';
import styled from 'styled-components';
import Playground from './Playground';
import * as components from '../../src/';
import ThemeProvider from '../../src/ThemeProvider';

const { Heading } = components;

const Body = styled.div`
  background: #FFFFFF;
`;

const filePathToComponentName = file =>
  file.replace(/\.\/(examples\/)?/, '').replace(/\.example$/, '');

const getExamples = () => {
  const req = require.context('./examples', true, /\.example$/);
  return req.keys().map((file) => {
    const code = req(file);
    const scope = [];

    let match;
    const componentNameRegex = /<([A-Z]\w*)[\s|>]/g;
    // eslint-disable-next-line
    while ((match = componentNameRegex.exec(code)) !== null) {
      scope.push({
        name: match[1],
        component: components[match[1]],
      });
    }

    return {
      name: filePathToComponentName(file),
      scope: scope.filter(x => x.component)
        .reduce((acc, cur) => ({
          ...acc,
          [cur.name]: cur.component,
        }), {}),
      code,
    };
  });
};

export default () => (
  <ThemeProvider>
    <Body>
      <Heading size={1}>Playground</Heading>
      {
        getExamples()
          // eslint-disable-next-line
          .sort((x, y) => x.name > y.name ? 1 : -1)
          .map(x => (
            <Playground
              key={x.name}
              code={x.code}
              scope={x.scope}
              collapsableCode
            />
          ))
      }
    </Body>
  </ThemeProvider>
);
