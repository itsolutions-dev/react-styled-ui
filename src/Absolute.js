import styled, { css } from './styledComponent';
import { getSizeOrDefault } from './utils/';

const Absolute = styled({
  component: 'div',
  style: css`
    position: absolute;
    ${props => props.top !== undefined ? `top: ${getSizeOrDefault(props.top, 0)};` : ''}
    ${props => props.bottom !== undefined ? `bottom: ${getSizeOrDefault(props.bottom, 0)};` : ''}
    ${props => props.left !== undefined ? `left: ${getSizeOrDefault(props.left, 0)};` : ''}
    ${props => props.right !== undefined ? `right: ${getSizeOrDefault(props.right, 0)};` : ''}
  `,
});

export default Absolute;
