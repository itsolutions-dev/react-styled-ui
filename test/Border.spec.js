import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import getTheme from '../src/getTheme';
import Border from '../src/Border';
import * as utils from '../src/utils/';

const theme = getTheme({
  palette: {
    defaultBorder: '#bababa',
  },
});

describe('Border', () => {
  test('should render a Border', () => {
    const component = ReactTestRenderer.create(<Border theme={theme} />);
    expect(component).toBeDefined();
  });

  test('should render a div tag', () => {
    const component = ReactTestRenderer.create(<Border theme={theme} />).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should have a border color', () => {
    const spy = jest.spyOn(utils, 'getBorder');
    const component = ReactTestRenderer.create(<Border theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-color', '#bababa');
    expect(spy).toHaveBeenCalled();
  });

  test('should have a the given border color', () => {
    const spy = jest.spyOn(utils, 'getBorder');
    const component = ReactTestRenderer.create(<Border color="blue" theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-color', 'blue');
    expect(spy).toHaveBeenCalled();
  });

  test('should handle border-top', () => {
    let component = ReactTestRenderer.create(<Border theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-top-width', '0');
    component = ReactTestRenderer.create(<Border top theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-top-width', '1px');
    const spy = jest.spyOn(utils, 'getSizeOrDefault');
    component = ReactTestRenderer.create(<Border top={20} theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-top-width', '20px');
    expect(spy).toHaveBeenCalled();
  });

  test('should handle border-bottom', () => {
    let component = ReactTestRenderer.create(<Border theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-bottom-width', '0');
    component = ReactTestRenderer.create(<Border bottom theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-bottom-width', '1px');
    const spy = jest.spyOn(utils, 'getSizeOrDefault');
    component = ReactTestRenderer.create(<Border bottom={20} theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-bottom-width', '20px');
    expect(spy).toHaveBeenCalled();
  });

  test('should handle border-left', () => {
    let component = ReactTestRenderer.create(<Border theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-left-width', '0');
    component = ReactTestRenderer.create(<Border left theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-left-width', '1px');
    const spy = jest.spyOn(utils, 'getSizeOrDefault');
    component = ReactTestRenderer.create(<Border left={20} theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-left-width', '20px');
    expect(spy).toHaveBeenCalled();
  });

  test('should handle border-right', () => {
    let component = ReactTestRenderer.create(<Border theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-right-width', '0');
    component = ReactTestRenderer.create(<Border right theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-right-width', '1px');
    const spy = jest.spyOn(utils, 'getSizeOrDefault');
    component = ReactTestRenderer.create(<Border right={20} theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-right-width', '20px');
    expect(spy).toHaveBeenCalled();
  });

  test('should have the given border-style', () => {
    const component = ReactTestRenderer.create(<Border borderStyle="dotted" theme={theme} />).toJSON();
    expect(component).toHaveStyleRule('border-style', 'dotted');
  });
});
