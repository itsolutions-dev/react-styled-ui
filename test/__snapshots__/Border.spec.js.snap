// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Border should render a Border 1`] = `
.c0 {
  border-style: solid;
  border-top-width: 0;
  border-bottom-width: 0;
  border-left-width: 0;
  border-right-width: 0;
  border-color: #bababa;
  color: inherit !important;
}

<div
  className="c0"
/>
`;
